Automattic/wp-calypso
# Framework: Upgrade to Babel 6

This pull request seeks to upgrade Calypso from Babel 5.8.12 to Babel 6.9.x. [Babel 6](http://babeljs.io/blog/2015/10/29/6.0.0/) is a significant departure in its approach, switching to a modular approach. As such, it requires opting in to most syntax conversions.

A significant number of the changed files are a consequence of the splitting of `babel/register` to a separate [`babel-register` module](https://babeljs.io/docs/usage/require/). Since most of our tests use the require hook, each of their Makefiles needed to be updated to reflect the updated compiler module.

__Testing instructions:__

Run `make test` from the root directory and verify that all tests pass.

Ensure that regressions should be found in navigating the application in the browser.
